File: LeetCode/Easy/0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cs
public class Solution {
    public string ReverseVowels(string s)
        => string.Create(s.Length, s, (span, input) =>
            {
                input.AsSpan().CopyTo(span);

                var left = 0;
                var right = input.Length - 1;

                while (left < right)
                {
                    while (left < right && !IsVowel(span[left])) left++;
                    while (left < right && !IsVowel(span[right])) right--;

                    if (left == right) return;

                    var tmp = span[left];
                    span[left] = span[right];
                    span[right] = tmp;
                    left++;
                    right--;
                    
                }
            }
        );
    
    private bool IsVowel(char c)
    => c switch
        {
            'a' or 'e' or 'i' or 'o' or 'u' or 'A'or 'E' or 'I' or 'O' or 'U' => true,
            _ => false
        };
}

File: LeetCode/Easy/0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        VOWELS = set("aeiouAEIOU")

        left = 0
        right = len(s) - 1
        result = list(s)

        while left < right:
            while left < right and result[left] not in VOWELS:
                left += 1
            
            while left < right and result[right] not in VOWELS:
                right -= 1
            
            if left < right:
                result[left], result[right] = result[right], result[left]
            
            left += 1
            right -= 1

        return "".join(result)